File: typescript/src/api/resources/onlineEvaluations/client/requests/PolicyInput.ts

/**
 * @example
 *     {
 *         evaluator: "evaluator",
 *         id: "id",
 *         revision: "revision",
 *         rubric: "rubric",
 *         samplePercent: 1
 *     }
 */
export interface PolicyInput {
    environment?: PolicyInput.Environment;
    evaluator: string;
    id: string;
    revision: string;
    rubric: string;
    samplePercent: number;
}

export namespace PolicyInput {
    export const Environment = {
            Production: "production",
            Staging: "staging",
            Development: "development"
        } as const;
    export type Environment = typeof Environment[keyof typeof Environment];
}

File: ruby/lib/Sikaru/online_evaluations/types/policy_input.rb
# frozen_string_literal: true

module Sikaru
  module OnlineEvaluations
    module Types
      class PolicyInput < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :environment, -> { Sikaru::OnlineEvaluations::Types::PolicyInputEnvironment }, optional: true, nullable: false

        field :evaluator, -> { String }, optional: false, nullable: false

        field :id, -> { String }, optional: false, nullable: false

        field :revision, -> { String }, optional: false, nullable: false

        field :rubric, -> { String }, optional: false, nullable: false

        field :sample_percent, -> { Integer }, optional: false, nullable: false, api_name: "samplePercent"
      end
    end
  end
end
